src/guides/protocol.md

```diff
@@ -0,0 +1,230 @@
+
+Name covenants are linked together in a chain, in the same way that transaction
+inputs and outputs transfer value in a chain. In addition, some covenants must
+identify the correct [name hash](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/covenants/rules.js#L1196)
+to be valid. There are
+[rules](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/covenants/rules.js#L1062)
+about the sequence in which covenants can be spent. A transaction can have more
+than one covenant, but the input and output indexes are linked.
+
+#### These covenant types negotiate the name auctions:
+
+| From --> | --> To     | Purpose   |
+|----------|------------|-----------|
+| `NONE`, `REDEEM`, `OPEN`   | `NONE`     | _Just sending money_ |
+| `NONE`, `REDEEM`, `OPEN`   | `OPEN`     | _Start an auction for an available name_ |
+| `NONE`, `REDEEM`, `OPEN`   | `BID`      | _Bid on an open auction for a name with a blinded value_ |
+| `BID`    | `REVEAL`   | _Reveal the amount that was bid on a name_ |
+| `REVEAL` | `REDEEM`   | _Sweep a losing auction bid back to wallet_ |
+| `NONE`   | `CLAIM`    | _Names in the Alexa top 100,000 are reserved for legacy DNS owners_ |
+| `CLAIM`, `REVEAL`  | `REGISTER` | _Update the DNS resource for an owned or won name (effectively burns coins)_ |
+
+Because Handshake uses
+[Vickrey Auctions](https://en.wikipedia.org/wiki/Vickrey_auction),
+the highest bidder pays
+[the second-highest bid.](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/blockchain/chain.js#L1077-L1083)
+In the event only one bid is submitted, that bidder can recover their entire bid
+(when they `REGISTER`) in essence, paying the second-highest bid of zero.
+
+Once a name has been `REGISTER`ed, it's value is locked forever, but the namestate
+can still be changed.
+
+#### These covenant types update "owned" names:
+
+| From --> | --> To     | Purpose   |
+|----------|------------|-----------|
+| `REGISTER`, `UPDATE`, `RENEW`, `TRANSFER`, `FINALIZE` | `UPDATE` | _Update the DNS resource_ |
+| `REGISTER`, `UPDATE`, `RENEW`, `TRANSFER`, `FINALIZE` | `RENEW` | _Renew name ownership before expiration_ |
+| `REGISTER`, `UPDATE`, `RENEW`, `FINALIZE` | `TRANSFER` | _Initiate ownership transfer to new Handshake address_ |
+| `TRANSFER` | `FINALIZE` | _Confirm ownership to new Handshake address (after lockup period)_ |
+| `REGISTER`, `UPDATE`, `RENEW`, `TRANSFER`, `FINALIZE`  | `REVOKE` | _Permanently burn name -- used when a key is compromised_ |
+
+Transfer and revocation actions are not allowed until
+[about two weeks](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/protocol/networks.js#L310)
+after the reveal period ends.
+
+Names must be renewed within
+[two years](https://github.com/handshake-org/hsd/blob/b32c27bbb05e69b86c0e80aea3ecc394c5589186/lib/protocol/networks.js#L250)
```

Comment:
Good find, I had thought it was 1 year.  The max number of names that could be theoretically supported by the blockchain is if every block is full of only renewal transactions

Reply:
👍